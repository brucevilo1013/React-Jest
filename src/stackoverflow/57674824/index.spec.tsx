import React from 'react';
import { SomeCompoennt } from './';
import { shallow, ShallowWrapper } from 'enzyme';

describe('SomeCompoennt', () => {
  const mockProps = {
    object: {
      objectId: '12'
    },
    fetchObjectDetails: jest.fn()
  };

  let wrapper: ShallowWrapper;
  beforeEach(() => {
    wrapper = shallow(<SomeCompoennt {...mockProps}></SomeCompoennt>);
    mockProps.fetchObjectDetails.mockClear();
  });

  test('should shallow render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should call reloadObjectDetails when objectId has changed', () => {
    const objectId = '123';
    const nextProps = { object: { objectId } };
    const spy = jest.spyOn(SomeCompoennt.prototype, 'componentWillReceiveProps');
    const reloadObjectDetailsSpy = jest.spyOn(SomeCompoennt.prototype as any, 'reloadObjectDetails');
    wrapper.setProps(nextProps);
    expect(reloadObjectDetailsSpy).toBeCalledWith(objectId);
    expect(spy).toBeCalledTimes(1);
    expect(mockProps.fetchObjectDetails).toBeCalledWith(objectId);
    expect(wrapper).toMatchSnapshot();
  });
});
