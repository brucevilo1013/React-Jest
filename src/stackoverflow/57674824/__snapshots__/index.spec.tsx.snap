// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SomeCompoennt should call reloadObjectDetails when objectId has changed 1`] = `
<div>
  123
</div>
`;

exports[`SomeCompoennt should shallow render correctly 1`] = `
<div>
  12
</div>
`;
